Sources/eppo/EppoClient.swift

```diff
@@ -51,32 +52,41 @@ public class EppoClient {
         sdkKey: String,
         host: String,
         assignmentLogger: AssignmentLogger? = nil,
-        assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
+        assignmentCache: AssignmentCache? = InMemoryAssignmentCache(),
+        configurationRequester: ConfigurationRequesterProtocol
     ) {
         self.sdkKey = sdkKey
         self.host = host
         self.assignmentLogger = assignmentLogger
         self.assignmentCache = assignmentCache
         
-        let httpClient = NetworkEppoHttpClient(baseURL: host, sdkKey: sdkKey, sdkName: "sdkName", sdkVersion: sdkVersion)
-        let configurationRequester = ConfigurationRequester(httpClient: httpClient)
         self.configurationStore = ConfigurationStore(requester: configurationRequester)
     }
     
     public static func initialize(
         sdkKey: String,
         host: String = "https://fscdn.eppo.cloud",
         assignmentLogger: AssignmentLogger? = nil,
-        assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
+        assignmentCache: AssignmentCache? = InMemoryAssignmentCache(),
+        forceReinitialize: Bool = false
```

Comment:
This method respects the singleton. For the use case of wanting to load the remote configuration after the static one, need to be able to create a new singleton with the sdk key.

Reply:
🔴 after another read I suspect there is an undesirable behavior for customers that perform a remote load after setting a static configuration in that the newly created singleton will not have remote configuration loaded and be blank for a short period of time.

I will take another pass at this implementation to cover it.

Reply:
What needs to happen here is the configuration be loaded from remote and only then the singleton be over-written.

Note to self: write a test for this case.

Reply:
I support what @aarsilv is saying.

I would say that we must preserve `EppoClient` identity in case the result of the first `initialize()` call is saved by the user. Calling `initialize` twice is very confusing as well.

I have implemented this feature recently in Python with an optional `initial_configuration` parameter: https://github.com/Eppo-exp/python-sdk/pull/70
I also suggest adding a similar `Configuration` abstraction as it's very likely that we'll allow users to initialize bandit configuration in the future as well.